### ISSUE TITLE
The clash got disconnected sometimes. 

### ISSUE THREAD

**Issue by maywzh** [Dreamacro/clash#111]
OS: **Windows 10 professional 1803**
Client: [Clash for windows](https://github.com/Fndroid/clash_for_windows_pkg)

The clash got disconnected sometimes. Restarting the client can solve it. 
This is the log.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x525b9a]

goroutine 11141 [running]:
github.com/Dreamacro/clash/constant.(*Metadata).String(0xc00031b340, 0xc00019ef00, 0xc000197c27)
	/home/travis/gopath/src/github.com/Dreamacro/clash/constant/metadata.go:44 +0x3a
github.com/Dreamacro/clash/tunnel.(*Tunnel).match(0xc00008c4b0, 0xc00031b340, 0x0, 0x0)
	/home/travis/gopath/src/github.com/Dreamacro/clash/tunnel/tunnel.go:160 +0x285
github.com/Dreamacro/clash/tunnel.(*Tunnel).handleConn(0xc00008c4b0, 0x8e4be0, 0xc00032da80)
	/home/travis/gopath/src/github.com/Dreamacro/clash/tunnel/tunnel.go:133 +0x405
created by github.com/Dreamacro/clash/tunnel.(*Tunnel).process
	/home/travis/gopath/src/github.com/Dreamacro/clash/tunnel/tunnel.go:88 +0x9a

**Comment by Dreamacro** [comment]
Got it, would fix next release

**Comment by Dreamacro** [comment]
Fixed in e30a628
